site/src/components/AuthProvider/AuthProvider.tsx

```diff
@@ -0,0 +1,36 @@
+import { useActor, useInterpret } from "@xstate/react"
+import { createContext, FC, PropsWithChildren, useContext } from "react"
+import { authMachine } from "xServices/auth/authXService"
+import { ActorRefFrom } from "xstate"
+
+interface AuthProviderContextValue {
+  authService: ActorRefFrom<typeof authMachine>
+}
+
+const AuthProviderContext = createContext<AuthProviderContextValue | undefined>(
+  undefined,
+)
+
+export const AuthProvider: FC<PropsWithChildren> = ({ children }) => {
+  const authService = useInterpret(authMachine)
+
+  return (
+    <AuthProviderContext.Provider value={{ authService }}>
+      {children}
+    </AuthProviderContext.Provider>
+  )
+}
+
+// The returned type is kinda complex to rewrite it
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types -- Read above
```

Comment:
I took a look at this locally...man that is complex lol. I asked in Matt's Discord and on Twitter. If I find a solution, I'll submit a followup PR.

Reply:
Yes, it is haha